### PR TITLE
Add Organizations

### PR DESCRIPTION
**Pull request by francisco-leal**
## Summary

Adds Organization Model that can hold teams and communities

### CODE REVIEW

File: app/controllers/api/v1/organizations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    if organization_params[:type].present?
+      organizations = organizations.where(type: org_param_to_type)
+    end
+
+    pagy, organizations = pagy(organizations, items: per_page)
+
+    render(
+      json: {
+        organizations: organizations,
```
Comment (francisco-leal) [inline app/controllers/api/v1/organizations_controller.rb:22]: We're missing the blueprint after we understand what we want for the Frontend.

File: app/controllers/api/v1/organizations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  def per_page
+    params[:per_page] || PER_PAGE
+  end
+
+  def organization_params
+    params.permit(:user_id, :type)
+  end
+
+  def org_param_to_type
```
Comment (francisco-leal) [inline app/controllers/api/v1/organizations_controller.rb:50]: This is just to simply the interface between frontend => backend

File: app/controllers/api/v1/organizations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
+class API::V1::OrganizationsController < ApplicationController
```
Comment (RubenSousaDinis) [inline app/controllers/api/v1/organizations_controller.rb:1]: This should be placed in the `PublicAPI` namespace

File: spec/requests/api/v1/organizations_spec.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
+require "rails_helper"
+
+RSpec.describe "Organizations", type: :request do
```
Comment (RubenSousaDinis) [inline spec/requests/api/v1/organizations_spec.rb:3]: We should have a swagger spec in order to generate the documentation

File: db/migrate/20230503105916_create_organizations.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class CreateOrganizations < ActiveRecord::Migration[7.0]
+  def change
+    create_table :organizations do |t|
```
[the remark this answers is not in the document]
Comment (shirleyodeng) [reply db/migrate/20230503105916_create_organizations.rb:3]: @RubenSousaDinis - do you want me to make all fields not nullable?

[the remark this answers is not in the document]
Comment (shirleyodeng) [reply db/migrate/20230503105916_create_organizations.rb:3]: I've added banner data and location and made the name and type not nullable. @RubenSousaDinis WDYT?

[the remark this answers is not in the document]
Comment (shirleyodeng) [reply db/migrate/20230503105916_create_organizations.rb:3]: And an UUID for the url?

[the remark this answers is not in the document]
Comment (francisco-leal) [reply db/migrate/20230503105916_create_organizations.rb:3]: We should add a slug yes @shirleyodeng, in regards to the social fields, you can add them as well please 🙏

**Pull request merged**